### PR TITLE
Add NLog dependency injection

### PR DESCRIPTION
**Pull request by tomasmuzas**

### CODE REVIEW

File: BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs [hunk 1]
```diff
@@ -9,11 +11,19 @@ namespace BackendApartmentReservation.Controllers
     [ApiController]
     public class ApartmentsController : ControllerBase
     {
+        private ICustomLogger<ApartmentsController> _logger;
```
Comment (andrejus-kuznecovas) [inline BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs]: ## Question
Possible to make it readonly? -Sonar suggests

File: BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs [hunk 2]
```diff
@@ -9,11 +11,19 @@ namespace BackendApartmentReservation.Controllers
...
+        {
+            _logger = logger;
+        }
+
         [HttpGet]
         [Route("apartments")]
         
         public async Task<IEnumerable<string>> Get()
         {
+            _logger.Info("Initial step");
```
Comment (andrejus-kuznecovas) [inline BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs]: ## Suggestion
Maybe logging should give info about which component is active? For example 
``` _logger.Info("ApartmentsController - Initial step"); ``` or something similar.

**Review by andrejus-kuznecovas — APPROVED** [review]

**Review by andrejus-kuznecovas — APPROVED** [review]

File: BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs (hunk 1 above)
Comment (tomasmuzas) [reply BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs]: Definitely :)

File: BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs (hunk 2 above)
Comment (tomasmuzas) [reply BackendApartmentReservation/BackendApartmentReservation/Controllers/ApartmentsController.cs]: This is already taken care of behind the scene. Thats why you dont have to worry about any meta-information

**Comment by tomasmuzas** [comment]
I am planning on changing implementation not to have my own wrappers which are useless.

**Review by andrejus-kuznecovas — APPROVED** [review]

**Pull request merged**